AdvLoggerPkg/Library/AdvancedLoggerLib/MmCore/AdvancedLoggerLib.c

```diff
@@ -88,29 +89,28 @@ AdvancedLoggerGetLoggerInfo (
 {
   EFI_HOB_GUID_TYPE    *GuidHob;
   ADVANCED_LOGGER_PTR  *LogPtr;
-  STATIC BOOLEAN       Initialized = FALSE;
 
-  if (!Initialized) {
-    Initialized = TRUE;   // Only one attempt at getting the logger info block.
+  if (!mInitialized) {
+    mInitialized = TRUE;   // Only one attempt at getting the logger info block.
 
     //
     // Locate the Logger Information block.
     //
-    GuidHob = GetFirstGuidHob (&gAdvancedLoggerHobGuid);
+    GuidHob = (EFI_HOB_GUID_TYPE *)GetFirstGuidHob (&gAdvancedLoggerHobGuid);
     if (GuidHob == NULL) {
       DEBUG ((DEBUG_ERROR, "%a: Advanced Logger Hob not found\n", __FUNCTION__));
     } else {
       LogPtr      = (ADVANCED_LOGGER_PTR *)GET_GUID_HOB_DATA (GuidHob);
       mLoggerInfo = ALI_FROM_PA (LogPtr->LogBuffer);
       if (mLoggerInfo != NULL) {
         mMaxAddress = LOG_MAX_ADDRESS (mLoggerInfo);
+        DEBUG ((DEBUG_INFO, "%a: LoggerInfo=%p\n", __FUNCTION__, mLoggerInfo));

```

Comment:
while you are changing this, may as well go to `__func__`. Also, this is the one instance where `%a` is not that helpful on its own because we have so many of these libs that print where logger info is. We should probable add which module it is too (that could be a separate PR adding to each lib)